guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -509,11 +513,9 @@ private RemoteRepository getRemoteRepoFromDeployment( DeploymentRepository repo
         Server server = settings.getServer( repo.getId() );
 
         if ( server != null ) {
-            remoteRepoBuilder.setAuthentication(
-                    new AuthenticationBuilder()
-                            .addUsername( server.getUsername() )
-                            .addPassword( server.getPassword() )
-                            .build() );
+            Authentication authentication = embedder.getMavenSession().getRepositorySession()
+                    .getAuthenticationSelector().getAuthentication( remoteRepoBuilder.build() );
```

Comment:
Hi, I'm a little concerned that this PR may not be using org.eclipse.aether correctly; the [Java Docs](http://download.eclipse.org/aether/aether-core/1.0.0/apidocs/org/eclipse/aether/DefaultRepositorySystemSession.html#getAuthenticationSelector()) state "Note that this selector is not used for remote repositories..." whereas we seem to be specifically using it to configure remotes' Authentication. Am I missing something?